### PR TITLE
AngularJS Filters

### PR DESCRIPTION
**Pull request by ItachiUchiha1998**

### CODE REVIEW

**Comment by ItachiUchiha1998** [comment]
Hope its what you were looking for..

File: README.md [hunk 1]
```diff
@@ -327,8 +327,82 @@ Templates in React are written inside the JavaScript file using the [JSX languag
...
+Built-in Filters are as follows : <br />
+  -`currency` Format a number to a currency format.<br />
+  -`date` Format a date to a specified format.<br />
+  -`filter` Select a subset of items from an array.<br />
+  -`json` Format an object to a JSON string.<br />
+  -`limitTo` Limits an array/string, into a specified number of elements/characters.<br />
+  -`lowercase` Format a string to lower case.<br />
+  -`number` Format a number to a string.<br />
+  -`orderBy` Orders an array by an expression.<br />
+  -`uppercase` Format a string to upper case.<br />
```
Comment (feimosi) [inline README.md]: Please remove all the `<br />`, markdown will handle that by itself.

File: README.md [hunk 2]
```diff
@@ -327,8 +327,82 @@ Templates in React are written inside the JavaScript file using the [JSX languag
 
 # Filters
 
-> TODO
-> Angular's pipe character (|) 
+### AngularJS
+
+AngularJS provides filters to transform data for transform of data.<br />
```
Comment (feimosi) [inline README.md]: This sentence sounds strange. Should be `AngularJS provides filters to transform data.`

File: README.md [hunk 3]
```diff
@@ -327,8 +327,82 @@ Templates in React are written inside the JavaScript file using the [JSX languag
...
+There are several built-in filters to use or you can make your own custom filters as well.<br />
+Filters can applied to view template using following syntax :  `{{ expression | filter }}`<br />
```
Comment (feimosi) [inline README.md]: Grammar correction: `Filters can be applied to view template using the following syntax:`

File: README.md [hunk 4]
```diff
@@ -327,8 +327,82 @@ Templates in React are written inside the JavaScript file using the [JSX languag
...
+<script>
+var app = angular.module('myApp', []);
+app.controller('caseCtrl', function($scope) {
+    $scope.txt = "Hello World!";
+});
+</script>
+```
+
+
+Code Snippet of Custom Filter :
```
Comment (feimosi) [inline README.md]: Please leave just one empty space between text / heading / code snippet. Markdown will ignore it either way.

File: README.md [hunk 5]
```diff
@@ -327,8 +327,82 @@ Templates in React are written inside the JavaScript file using the [JSX languag
...
+  class MyFilter {
+   
+  static filter( value ){
+    return value.toLowerCase();
+  }
+   
+}
+ 
+export default MyFilter.filter;
+```
```
Comment (feimosi) [inline README.md]: This code is actually unnecessary. `lowercase` is built-in.

File: README.md [hunk 6]
```diff
@@ -327,8 +327,82 @@ Templates in React are written inside the JavaScript file using the [JSX languag
...
+
+</div>
+
+<script>
+var app = angular.module('myApp', []);
+app.controller('caseCtrl', function($scope) {
+    $scope.txt = "Hello World!";
+});
+</script>
+```
```
Comment (feimosi) [inline README.md]: Just
`<h1>{{txt | lowercase}}</h1>`
for this example would be enough.

File: README.md [hunk 7]
```diff
@@ -327,8 +327,82 @@ Templates in React are written inside the JavaScript file using the [JSX languag
...
+
+```
+angular.module('myReverseFilterApp', [])
+.filter('reverse', function() {
+  return function(input, uppercase) {
+    input = input || '';
+    var out = '';
+    for (var i = 0; i < input.length; i++) {
+      out = input.charAt(i) + out;
+    }
```
Comment (feimosi) [inline README.md]: This `for` could be written as:
`const out = input.split('').reverse().join('')`

File: README.md [hunk 8]
```diff
@@ -327,8 +327,82 @@ Templates in React are written inside the JavaScript file using the [JSX languag
...
+    if (uppercase) {
+      out = out.toUpperCase();
+    }
+    return out;
+  };
+})
+.controller('MyController', ['$scope', 'reverseFilter', function($scope, reverseFilter) {
+  $scope.greeting = 'hello';
+  $scope.filteredGreeting = reverseFilter($scope.greeting);
+}]);
```
Comment (feimosi) [inline README.md]: This controller can be omitted.

**Review by feimosi — CHANGES_REQUESTED** [review]
Thanks! I've put some comments, please check.

**Comment by ItachiUchiha1998** [comment]
I have made the changes please check them.

Regards

**Comment by feimosi** [comment]
Looks good, thank you! 🥇

**Pull request merged**